### ISSUE TITLE
Is it possible to learn neuralode for stiff problem of Robertson's problem?

### ISSUE THREAD

**Issue by jiweiqi** [SciML/DiffEqFlux.jl#367]
Hi,

I adapted the example neural ode code at https://diffeqflux.sciml.ai/dev/examples/neural_ode_flux/ to stiff ode problem of [Robertson's problem](https://docs.sciml.ai/latest/tutorials/advanced_ode_example/#Handling-Mass-Matrices-1). It seems that it doesn't work. Couple questions about it:

1. Is there anyone who has successfully learned Robertson's problem using neural ode?
2. What's the math intuition behind of [Enforcing Physical Constraints via Universal Differential-Algebraic Equations](https://diffeqflux.sciml.ai/dev/examples/physical_constraints/). Is it kind of preconditioning? Where comes the physics constrains of `y_1 + y_2 + y_3 = 1`?
3. Any general suggestions in learning neural code for stiff systems.
3. Is there any theoretical/empirical work that analysis the difficulties for learning neural ode in the stiff system? For example, it is possible that there is some equivalence between stochastic gradient descent (SGD) and the explicit method. Such that training neural ode using SGD is fundamentally not working ...

Thanks in advance

**Comment by ChrisRackauckas** [comment]
>Is there anyone who has successfully learned Robertson's problem using neural ode?

https://diffeqflux.sciml.ai/dev/examples/physical_constraints/ is an example. I think that in general, you're going to need extra information like that in order to make it work.

>Where comes the physics constrains of y_1 + y_2 + y_3 = 1?

It's an equivalent form for the Robertson equations (https://docs.sciml.ai/latest/tutorials/advanced_ode_example/#Handling-Mass-Matrices-1) since you have one chemical moving between 3 states.

>Is there any theoretical/empirical work that analysis the difficulties for learning neural ode in the stiff system? For example, it is possible that there is some equivalence between stochastic gradient descent (SGD) and the explicit method. Such that training neural ode using SGD is fundamentally not working ...

I don't know if there's anything published, but we have a lot of work in progress on this exact topic. If you want to join the research conversation, let me know.

**Comment by jiweiqi** [comment]
Hi @ChrisRackauckas, I will try the constraints. It will be great if I can participate in the conversation. I am working on this topic in the context of modeling combustion chemistry. Stiffness is something we have to tackle. I am from MechE and my e-mail is weiqiji_at_mit.edu.

**Comment by jiweiqi** [comment]
Hi @ChrisRackauckas, jus for making sure. Does the example https://diffeqflux.sciml.ai/dev/examples/physical_constraints/ correctly learn the neural network yet? i.e., reproducing the species concentrations correctly. It seems not correct in my test.

BTW, the example is for the time domain of [0, 1]. I am trying to increase it to [0, 1e5].

**Comment by ChrisRackauckas** [comment]
It does on the short time span. The longer time span is much harder.

**Comment by jiweiqi** [comment]
Got it. Look forward to talking to you in more detail.
